pkg/controller_helper/backend/sglang.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backend
+
+import (
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	coreapi "inftyai.com/llmaz/api/core/v1alpha1"
+	inferenceapi "inftyai.com/llmaz/api/inference/v1alpha1"
+	"inftyai.com/llmaz/pkg"
+	datasource "inftyai.com/llmaz/pkg/controller_helper/datasource"
+)
+
+var _ Backend = (*SGLANG)(nil)
+
+type SGLANG struct{}
+
+const (
+	sglang_image_registry = "lmsysorg/sglang"
+)
+
+func (v *SGLANG) Name() inferenceapi.BackendName {
+	return inferenceapi.SGLANG
+}
+
+func (v *SGLANG) Image(version string) string {
+	return sglang_image_registry + ":" + version
+}
+
+func (v *SGLANG) DefaultVersion() string {
+	return "v0.2.9-cu121"
+}
+
+func (v *SGLANG) DefaultResources() inferenceapi.ResourceRequirements {
+	return inferenceapi.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("4"),
+			corev1.ResourceMemory: resource.MustParse("8Gi"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("4"),
+			corev1.ResourceMemory: resource.MustParse("8Gi"),
+		},
+	}
+}
+
+func (v *SGLANG) DefaultCommands() []string {
+	return []string{"python3", "-m", "sglang.launch_server"}
+}
+
+func (v *SGLANG) DefaultArgs(model *coreapi.Model) []string {
+	dataSource := datasource.NewDataSourceProvider(model)
+	return []string{
+		"--model-path", dataSource.ModelPath(),
+		"--served-model-name", dataSource.ModelName(),
```

Comment:
Sorry, does sglang support this flag?

Reply:
https://github.com/sgl-project/sglang/blob/ebf69964cd7e5af9a079eb430ecdd7f67e8566d0/python/sglang/srt/server_args.py#L212
We can find this flag here

Reply:
That's great